crates/driver/src/domain/liquidity/mod.rs

```diff
@@ -32,3 +39,18 @@ impl PartialEq<usize> for Id {
         self.0 == *other
     }
 }
+
+/// Data tied to a particular liquidity instance, specific to the type of
+/// liquidity.
+///
+/// This contains relevant data for encoding interactions for the given
+/// liquidity, as well as state required by the solver engine.
+#[derive(Debug, Clone)]
+pub enum Data {
+    UnswapV2(uniswap::v2::Pool),
+    UnswapV3(uniswap::v3::Pool),
+    BalancerV2Stable(balancer::stable::Pool),
+    BalancerV2Weighted(balancer::weighted::Pool),
+    Swapr(swapr::Pool),
+    ZeroEx(zeroex::LimitOrder),
+}
```

Comment:
This is not meant to be private data, just public data that is pool dependent (where the other fields in `Liquidity` are shared).

> In this case I'd personally call this `Kind` rather than `Data`.

This makes a lot of sense, and is more "Rust-y" (whatever that means), will change.

Reply:
8771a8147b99a1666aa96985ef11e085a32c835b